gm4_vertical_rails/data/gm4_vertical_rails/functions/rail/check_direction.mcfunction

```diff
@@ -0,0 +1,9 @@
+# Checks if Vertical Rails go up or downward, prioritising up
+# @s = minecart, touching a Vertical Rail (ladder)
+# at @s
+# run from process
+
+tag @s add gm4_vert_rail_momentum
+
+execute as @s[tag=!gm4_vert_rail_down] at @s if block ~ ~1 ~ ladder run tag @s add gm4_vert_rail_up
```

Comment:
```suggestion
execute if entity @s[tag=!gm4_vert_rail_down] if block ~ ~1 ~ ladder run tag @s add gm4_vert_rail_up
```
The function is already being called `as @s at @s` and no changes to the position of `@s` are made during the function, therefore `at @s` may be omitted to make this function run faster. Usage of `if entity @s` will also speed up the function in comparison to reselecting the frame of reference to `as @s`.